markers/sauce.py

```diff
@@ -18,3 +19,11 @@ def pytest_configure(config):
             config.option.markexpr = 'sauce and ({})'.format(config.option.markexpr)
         else:
             config.option.markexpr = 'sauce'
+
+
+def pytest_runtest_setup(item):
+    browser.ensure_browser_open(item)
+
+
+def pytest_runtest_teardown():
```

Comment:
I think perhaps the better thing here to do is to inspect the nextitem and see if we have a different browser, this way we can get the parallelizer to try to put all tests for the same browser on one appliance/slave.